.github/workflows/test.yml

```diff
@@ -0,0 +1,92 @@
+name: test
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches:
+      - develop
+jobs:
+  unit-test-job-id:
+    name: test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        test-name: [
+          'test-holochain',
+          'test-deps',
+          'test-misc',
+          'test-bin'
+        ]
+        os: [
+          ubuntu-latest,
+          macos-latest,
+          macos-13,
+          windows-latest,
+        ]
+        toolchain: [
+          '1.77.2',
+        ]
+    steps:
+      - name: free disk space
+        if: matrix.os == 'ubuntu-latest'
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: go toolchain
+        uses: actions/setup-go@v5
+        with:
+          go-version: '=1.20.0'
+
+      - name: rust toolchain
```

Comment:
Shouldn't toolchain selection happen automatically based on the toolchain.toml without needing selection in CI?

Reply:
Good point, let's test that out: https://github.com/holochain/holochain/pull/3894/commits/6b9830ec5b9a5b2d8971ad64244155d5e6879a77